### ISSUE TITLE
Adam vs AdamW 

### ISSUE THREAD

**Issue by qubit0** [google-deepmind/optax#417]
Pytorch's implementation of Adam takes weight_decay as an option. It also has AdamW. Most of the papers, seem to use Adam with weight_decay instead of AdamW. Is that the correct thing to do? 

However, optax's implementation of Adam doesn't take weight_decay. AdamW handles weight_decay. When I am trying to write a pytroch equivalent in jax, what optimizer should I use?

**Comment by mtthss** [comment]
In optax we have two separate optimisers for `adam` and `adamw`, respectively.
If you are interested in using weight decay with adam you should use `adamw`.

**Comment by qubit0** [comment]
Doyou think the pytorch's implementation of `Adam` with weight_decay is incorrect?

If I use optax's `AdamW` then I can't reproduce the result from the paper that used pytorch's `Adam` with weight_decay.  So, I was confused.

**Comment by mtthss** [comment]
optax adamw is equivalent to pytorch adamw

If you use pytorch adam (without the `w`) with weight_decay!=0 you are not doing adamw.
It adds an L2 penalty that also pushes the weights down but in a subtly different and worse way
(this is extensively discussed in the adamw that shows the *right* way of doing weight decay with adam: i.e. adamw)

**Comment by mtthss** [comment]
More recent papers usually adamw instead of adam with the L2 penalty
(adamw is quite recent so older papers might still use pytorch adam with weight_decay!=0)

**Comment by qubit0** [comment]
Thank you @mtthss

**Comment by jenkspt** [comment]
Hi I'm trying to apply l2 penalty to specific parameters like so:

```python
    optimizer = optax.chain(
        optax.clip_by_global_norm(grad_clip),
        # Apply weight decay excluding bias and embedding parameters
        optax.add_decayed_weights(weight_decay, mask=param_decay_mask(params)),
        optax.adamw(learning_rate, weight_decay=0.0),
    )
```

does `add_decayed_weights` + `adamw` perform the correctly adjusted parameter update?
